### ISSUE TITLE
/tpp sometimes spawns you inside blocks

### ISSUE THREAD

**Issue by zrisher** [midspace/Space-Engineers-Admin-script-mod#74]
If a player is in a cockpit or cryotube and you `/tpp` to them you will spawn directly behind that block. Often this space is occupied by another block, killing you instantly if the grid is moving or within a couple seconds if you are unable to wiggle your way out.

It would be nice if `/tpp` checked for free space into which to teleport you and adjusted the destination if the default was not free.

**Comment by midspace** [comment]
Player to Player teleport already provides for safety checks.
It's the Player to Player in cockpit that needs safety checks.

Support.MovePlayerToCockpit() needs to be expanded, to uses the safely parameter the same as Support.MovePlayerToPlayer().

I actually haven't tested Cyro chamber occupation. :(
A quick check of the logic shows it should work in the /tpp call.
We should rename MovePlayerToCockpit() to something a little more accurate. Probably MovePlayerToCube().


(Sorry these are development notes just as much as a response, in case someone else if going to work on it.)

**Comment by zrisher** [comment]
Yep that's great, I just got back from a big trip and have lots to catch up on, but I do plan on doing some dev for you guys eventually, and this stuff is good to have for other incoming devs as well.

**Comment by midspace** [comment]
This should be fixed, although Cyro Chambers need to be tested.

**Comment by zrisher** [comment]
Rgr, will test and report back.
